DetectorDescription/DDCMS/plugins/DDTrackerPhiAltAlgo.cc

```diff
@@ -0,0 +1,78 @@
+#include "DD4hep/DetFactoryHelper.h"
+#include "DetectorDescription/DDCMS/interface/DDCMSPlugins.h"
+
+using namespace std;
+using namespace dd4hep;
+using namespace dd4hep::cms;
+
+static long algorithm(Detector& /* description */,
+                      ParsingContext& ctxt,
+                      xml_h e,
+                      SensitiveDetector& /* sens */)
+{
+  Namespace     ns(ctxt, e, true);
+  AlgoArguments args(ctxt, e);
+  Volume        mother      = ns.volume(args.parentName());
+  Volume        child       = ns.volume(args.childName());
+  double        tilt        = args.value<double>("Tilt");        //Tilt of the module
+  double        startAngle  = args.value<double>("StartAngle");  //offset in phi
+  double        rangeAngle  = args.value<double>("RangeAngle");  //Maximum range in phi
+  double        radiusIn    = args.value<double>("RadiusIn");    //Inner radius
+  double        radiusOut   = args.value<double>("RadiusOut");   //Outer radius
+  double        zpos        = args.value<double>("ZPosition");   //z position
+  int           number      = args.value<int>("Number");         //Number of copies
+  int           startCopyNo = args.find("StartCopyNo") ? args.value<int>("StartCopyNo") : 1; //Start copy number
+  int           incrCopyNo  = args.find("IncrCopyNo")  ? args.value<int>("IncrCopyNo")  : 1; //Increment in copy number
+
+  LogDebug("TrackerGeom") << "Parent " << mother.name() << "\tChild " << child.name() << " NameSpace " << ns.name;
+  LogDebug("TrackerGeom") << "Parameters for positioning-- Tilt " << tilt 
+                          << "\tStartAngle " << startAngle/CLHEP::deg 
+                          << "\tRangeAngle " << rangeAngle/CLHEP::deg 
+                          << "\tRin " << radiusIn << "\tRout " << radiusOut 
+                          << "\t ZPos " << zpos << "\tCopy Numbers " << number 
+                          << " Start/Increment " << startCopyNo << ", " 
+                          << incrCopyNo;
+
+  if (number > 0) {
+    double theta  = 90.*CLHEP::deg;
+    double dphi;
+    if (number == 1 || fabs(rangeAngle-360.0*CLHEP::deg)<0.001*CLHEP::deg) 
+      dphi = rangeAngle/number;
+    else
+      dphi = rangeAngle/(number-1);
+    int copyNo = startCopyNo;
+
+    for (int i=0; i<number; i++) {
+      double phi  = startAngle + i*dphi;
+      double phix = phi - tilt + 90.*CLHEP::deg;
```

Comment:
90.*CLHEP::deg;  this is simply ridiculous...
please use angles in radiant

Reply:
@VinInn - I agree, but it's one to one port of existing original Tracker code:
https://cmssdt.cern.ch/lxr/source/Geometry/TrackerCommonData/plugins/DDTrackerPhiAltAlgo.cc

Reply:
I see.
Then let's schedule it for a migration when some personpower will be available

Reply:
@VinInn - Yes, the right order is to improve current code, validate, then migrate, then validate again. The first can go in parallel when you'll find the personpower. You (e.g. Tracker group) will certainly keep the ownership of the code in its current location and migrate it to DD4Hep later, replacing the algorithms in this package.
This package is a merely playground for DD4Hep evaluation.